onnxruntime/test/python/transformers/test_sparse_attention.py

```diff
@@ -820,10 +818,66 @@ def run_relevance_past(self, sm: int, device, do_rotary: bool):
                     config.dtype = torch.bfloat16
                     self.run_one_relevance_test(config)
 
+    def run_relevance_no_past_128k(self, sm: int, device):
+        """Test kernel could support up to 128K sequence length."""
+        for seq_len in [131072]:
+            for packed_qkv in [False, True]:
+                config = SparseAttentionConfig(
+                    batch_size=1,
+                    sequence_length=seq_len,
+                    max_sequence_length=131072,
+                    past_sequence_length=0,
+                    num_heads=1,
+                    kv_num_heads=1,
+                    head_size=128,
+                    sparse_block_size=64,
+                    num_layout=1,
+                    local_blocks=2048,  # use dense to compare with GQA
+                    vert_stride=8,
+                    softmax_scale=None,
+                    device=device,
+                    is_packed_qkv=packed_qkv,
+                )
+                self.run_one_relevance_test(config)
+
+                if sm >= 80 and not packed_qkv:
+                    config.dtype = torch.bfloat16
+                    self.run_one_relevance_test(config)
+
```

Comment:
Can we add a unit test that runs `SparseAttention` with `do_rotary = true`? This is related to [this comment](https://github.com/microsoft/onnxruntime/pull/20609#discussion_r1594401320). We should also have a similar test for `GroupQueryAttention` with `do_rotary = true`.

Reply:
We do have test for do_rotary, but not the strange design of using different rotary cache for different length (that will need update the operator spec).